### PR TITLE
Feature/fajar/create endpoint view all instructor after register

### PR DESCRIPTION
**Pull request by fajartd02**
This branch create for 2 task
Issues:
https://trello.com/c/EiSipsMt/41-be-create-an-endpoint-to-view-all-instructors-registrations
https://trello.com/c/R0pWZJql/28-be-create-an-endpoint-to-verify-reject-instructor-registration

Step To reproduce
Issues 1:
- HIT GET http://localhost:8080/approval/register with auth, pastikan sudah register new instructor
- Akan dapat response:
```
{
    "status": "success",
    "data": [
        {
            "id": 10,
            "email": "dwdwdw@gmail.com",
            "password": "$2b$10$kq8qrD3KO8KEYL.BJvhpvOLiLEfwTKe4oKpV9UpLE/9TXU9eIt3hy",
            "name": "instructor",
            "role": 1,
            "isVerified": false,
            "refreshToken": null
        }
    ],
    "message": "Successfully get all new Instructor"
}
```

Issues 2:
- HIT POST http://localhost:8080/approval/:userId?action=#query
- Untuk querynya terdiri atas 2, approve || reject
- Sehingga untuk approve anda gunakan http://localhost:8080/approval/:userId?action=approve
- Untuk reject, http://localhost:8080/approval/:userId?action=reject

Contoh response berhasil:
*Approve*
`
{
    "status": "success",
    "message": "Successfully approve new instructor"
}`

*Reject*
`
{
    "status": "success",
    "message": "Successfully approve new instructor"
}
`

### CODE REVIEW

File: src/routes/index.ts [hunk 1]
```diff
@@ -21,4 +22,10 @@ router.post('/courses/:courseId/enroll', authenticate('ACCESS'),
 // Course
 router.post('/courses', authenticate('ACCESS'), courseController.add);
 
+// Approval
+router.get('/approval/register', authenticate('ACCESS'),
+    approvalController.getAllNewInstructor);
+router.post('/approval/:id', authenticate('ACCESS'),
```
Comment (rahmatsyifana12) [inline src/routes/index.ts]: ini ganti pake :userId aja biar ga bingung

File: src/services/approval.service.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        const instructorNotVerified = User.findBy({
+            isVerified: false,
+            role: UserRole.INSTRUCTOR
+        });
+
+        if (!instructorNotVerified) {
+            throw Errors.USER_NOT_FOUND;
+        }
```
Comment (rahmatsyifana12) [inline src/services/approval.service.ts]: kalo ga ada datanya, return array kosong aja

File: src/services/approval.service.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        const status = await this
+            .authorizationNewInstructor(instructor, action);
+        return status;
+    }
+
+    async authorizationNewInstructor(instructor: User, action: string) {
+        if (action === 'approve') {
```
Comment (rahmatsyifana12) [inline src/services/approval.service.ts:49]: bikin action type kayanya lebih bagus (just saran).

**Review by rahmatsyifana12 — CHANGES_REQUESTED** [review]

File: src/routes/index.ts (hunk 1 above)
Comment (fajartd02) [reply src/routes/index.ts]: ok bang

File: src/services/approval.service.ts (hunk 2 above)
Comment (fajartd02) [reply src/services/approval.service.ts]: okeh

**Pull request merged**